server/helper.ts

```diff
@@ -107,6 +107,30 @@ function getSearchkitRouter() {
         similars = [];
       }
 
+      if(source){
+        let otherLangPublications: Awaited<ReturnType<typeof elasticsearch.getRelatedPublications>>;
+        try {
+          // Get related publications from all other indices
+          otherLangPublications = await elasticsearch.getRelatedPublications(req.params.id, 1000, req.params.index);
+        } catch (e) {
+          otherLangPublications = [];
+          elasticsearchErrorHandler(e, res);
+        }
+
+        const selectedLang = req.params.index.split('_')[1];
+        if(source.relatedPublications){
+          // Add lang to the related publications of selected language
+          source.relatedPublications.forEach(publication => {
+            publication.lang = selectedLang;
+          });
+          // Add related publications in other languages
+          source.relatedPublications.push(...otherLangPublications);
+        } else {
+          // Replace original relatedPublications with related publications found in other languages
+          source.relatedPublications = otherLangPublications;
+        }
```

Comment:
This will only add other language related publications if the original language source has related publications defined. Is this intentional?

Reply:
Shouldn't this be fine unless I'm missing something? Since ```source.relatedPublications``` will be ```otherLangPublications``` if there's no related publications in original language. Technically if ```source.relatedPublications``` is empty array then it is still truthy and the ```otherLangPublications```will be pushed to be a part of it but that's also the same end result.

On an unrelated note, I just noticed that FSD actually has same related publications for en and fi for a lot of studies which means they will show up twice. Not sure what the cleanest solution for this is but maybe just checking if the same title title(+holdings) already exist and then keeping the one that was in first even though the language tag might then be incorrect.

Reply:
If the array is undefined, which may be the case as the raw results from ES haven't been normalised using `metadata.ts -> getStudyModel()`, then `otherLangPublications` won't be copied

Reply:
Then it should execute ```else``` block and set ```source.relatedPublications``` as ```otherLangPublications```. The comment for that could say "Set source.relatedPublications to related publications found in other languages" since it's not really "replacing" anything (besides undefined for example).

Reply:
Yes. I didn't see that. Approving this now.